### PR TITLE
Stack coord-norm fix + mirror-aug + SW=2.0 on 5L/256d [Wave18]

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Two independently validated improvements — (1) coordinate normalization fix (fern PR #409) and (2) mirror augmentation + surface loss weight 2.0 (tanjiro PR #443) — should stack. Together they address different aspects of the learning problem: coord-norm improves the geometric input representation, while mirror-aug+SW=2.0 improves data diversity and upweights the binding wall-shear constraint. Neither experiment tested this combination.

Evidence:
- fern #409 coord-norm fix: ep18=7.4516% and still improving at -0.082pp/ep; ep25 projection ~6.88% (potential new baseline beater vs 6.9549% current best)
- tanjiro #443 mirror-aug+SW=2.0: ep15=8.2789% PASS; wsy/wsz trend accelerating ep12→ep15

The coord-norm fix corrects the normalization of 3D coordinates passed into the Fourier PE layer, ensuring the input distribution matches the expected range. Mirror augmentation (y-axis flip p=0.5 with wsy/wsz sign-flip) doubles effective training data by exploiting the bilateral symmetry of aerodynamic shapes. SW=2.0 upweights the surface loss which targets the binding wsy/wsz metrics. All three are likely orthogonal: better geometry encoding + more data + stronger surface supervision.

## Instructions

This experiment requires **three code-level changes** to `target/train.py` (or the supporting model/dataset files) plus one CLI flag. Read fern's branch (`fern/coord-norm-fix-5l256d-wave13`, see PR #409) and tanjiro's branch (`tanjiro/mirror-aug-sw2-on-5l256d-wave14`, see PR #443) to understand exactly what each change does, then combine them:

### Change 1: Coordinate normalization fix (from fern PR #409)
Look at the diff on fern's branch vs bengio. Apply the same coordinate normalization fix to the dataset/dataloader code. The fix ensures 3D point coordinates are properly normalized before being passed to the FourierEmbed positional encoding layer.

### Change 2: Mirror augmentation (from tanjiro PR #443)
Look at the diff on tanjiro's branch vs bengio. Apply the same y-axis mirror augmentation:
- During training, randomly flip point clouds along the y-axis with probability p=0.5
- When flipping y-coordinates: multiply y-coords by -1
- When flipping, also flip the sign of `wall_shear_y` (wsy) and `wall_shear_z` (wsz) labels (physical symmetry: y-flip reverses lateral shear directions)
- Do NOT flip surface_pressure or volume_pressure — those are symmetric under y-flip

### Change 3: Surface loss weight (CLI flag)
Add `--surface-loss-weight 2.0` to upweight the surface loss relative to volume loss.

### Run command
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --surface-loss-weight 2.0 \
  --grad-clip-norm 1.0 \
  --no-compile-model \
  --epochs 50 \
  --wandb-group bengio-wave18 \
  --wandb-name senku-stack-coord-norm-mirror-sw2-5l256d-tmax50
```

### Kill thresholds
- ep5: abupt > 13.0% → kill
- ep10: abupt > 10.5% → kill
- ep15: abupt > 8.0% → kill (this is the critical gate — both component experiments clear it)
- ep20: abupt > 7.5% → kill
- ep25: target < 6.95% (current baseline). If not on track to beat baseline by ep25, report and we'll decide.

Report metrics every 5 epochs in a PR comment with the full axis breakdown (surface_pressure, wall_shear_y, wall_shear_z, volume_pressure, abupt).

## Baseline

**Current best: PR #174 (alphonse), W&B run `vu4jsiic`, 5L/256d/4H + FourierPE + T_max=50**

| Metric | Current Best (val) | AB-UPT Target |
|--------|-------------------|--------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | 4.51 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644 | 3.82 |
| `val_primary/volume_pressure_rel_l2_pct` | **3.9361** ✓ | 6.08 |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345 | 3.65 |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766 | 3.63 |

Beat 6.9549% on abupt to establish a new baseline. Wall shear y/z are the binding constraints — the combination of coord-norm + mirror-aug + SW=2.0 directly targets these.

**Component experiment benchmarks at ep18:**
- fern #409 (coord-norm only): 7.4516%
- tanjiro #443 (mirror-aug+SW=2.0 only): ~8.2% at ep15, projecting well

**Reproduce baseline:**
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --wandb-group bengio-wave9
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: @senku — Wave 18 PR assigned. Your pod is 1/1 READY.

**Task:** Stack coord-norm fix + mirror-aug (--mirror-aug-prob 0.5) + surface loss weight 2.0 on the 5L/256d+FourierPE baseline. This is the highest-priority hypothesis: tanjiro's mirror-aug run is showing the strongest results at ep~15 (abupt 8.28%) and this stacks the coord-norm fix on top.

Please ACK within 1 hour and post your W&B run ID once started. If you have any blockers or questions, post now.

**Comment by morganmcg1** [comment]
ADVISOR: @senku — This is a follow-up nudge. Your Wave 18 assignment was posted at 03:44Z. Over 1.5 hours have elapsed with no ACK. Please confirm you have received the task and post your ACK + estimated start time within the next 30 minutes. If you have a blocker, describe it now.

Task reminder: Stack coord-norm fix + mirror-aug (--mirror-aug-prob 0.5) + surface loss weight 2.0 on the 5L/256d+FourierPE baseline. 5L/256d/4H, fourier_pe=True, T_max=50, no-EMA, 50 epochs, 4 GPU, group=bengio-wave18.

If no response in 30 minutes, this PR will be reassigned.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #493 — no ACK received within the 30-minute window after nudge at 05:24Z. Pod shows 1/1 READY but no response. This assignment will be reassigned to another student. @senku: if you come back online and want to run this experiment, please open a fresh PR.
